apps/server/src/modules/federal-state/repo/federal-state.repo.ts

```diff
@@ -1,41 +1,72 @@
 import { EntityName } from '@mikro-orm/core';
-import { EntityManager } from '@mikro-orm/mongodb';
 import { Injectable } from '@nestjs/common';
+import { BaseRepo } from '@shared/repo/base.repo';
 import { FederalStateDO } from '../domainobject';
-import { FederalStateEntity } from '../entity';
+import { County, FederalStateEntity } from '../entity';
+import { IFederalStateCreate } from '../interface';
 
 @Injectable()
-export class FederalStateRepo {
-	constructor(private readonly em: EntityManager) {}
-
+export class FederalStateRepo extends BaseRepo<FederalStateEntity> {
 	get entityName(): EntityName<FederalStateEntity> {
 		return FederalStateEntity;
 	}
 
 	async findByName(name: string): Promise<FederalStateDO> {
-		const federalStateEntity = await this.em.findOneOrFail(FederalStateEntity, { name });
-		const federalState = this.mapEntityToDomainObject(federalStateEntity);
+		// const federalStateEntity = await this.findOneOrFail(FederalStateEntity, { name });
+		const federalStateEntity = await this._em.findOneOrFail(FederalStateEntity, { name });
+		const federalState = this.mapFederalStateEntityToDomainObject(federalStateEntity);
 		return federalState;
 	}
 
 	async findAll(): Promise<FederalStateDO[]> {
-		const federalStateEntities = await this.em.find(FederalStateEntity, {});
+		const federalStateEntities = await this._em.find(FederalStateEntity, {});
+		await this._em.populate(federalStateEntities, ['counties']);
 		const federalStates = federalStateEntities.map((federalStateEntity) =>
-			this.mapEntityToDomainObject(federalStateEntity)
+			this.mapFederalStateEntityToDomainObject(federalStateEntity)
 		);
 		return federalStates;
 	}
 
-	mapEntityToDomainObject(entity: FederalStateEntity): FederalStateDO {
+	async deleteFederalState(federalStateId: string): Promise<FederalStateDO> {
+		const federalStateEntity = await this.findById(federalStateId);
+		await this.delete(federalStateEntity);
+		const federalState = this.mapFederalStateEntityToDomainObject(federalStateEntity);
+		return federalState;
```

Comment:
delete federal state should not return the federal state again.... it was just deleted